webpack.config.js

```diff
@@ -363,43 +363,48 @@ async function initializeModuleFederation() {
-                aliases: {},
-            };
-        }
+        const remotes = {};
 
-        // Wait a second for product dev servers to start up if they were started at the same time as this one
-        await new Promise((resolve) => setTimeout(resolve, 1000));
+        if (process.env.MM_DONT_INCLUDE_PRODUCTS) {
+            if (DEV) {
+                // For development, identify which product dev servers are available
 
-        // For development, identify which product dev servers are available
-        const productsFound = await Promise.all(products.map((product) => isWebpackDevServerAvailable(product.baseUrl)));
+                // Wait a second for product dev servers to start up if they were started at the same time as this one
+                await new Promise((resolve) => setTimeout(resolve, 1000));
 
-        const remotes = {};
-        const aliases = {};
+                const productsFound = await Promise.all(products.map((product) => isWebpackDevServerAvailable(product.baseUrl)));
+                for (let i = 0; i < products.length; i++) {
+                    const product = products[i];
+                    const found = productsFound[i];
 
-        for (let i = 0; i < products.length; i++) {
-            const product = products[i];
-            const found = productsFound[i];
+                    if (found) {
+                        console.log(`Product ${product.name} found, adding as remote module`);
 
-            if (found) {
-                console.log(`Product ${product.name} found, adding as remote module`);
-
-                remotes[product.name] = `${product.name}@${product.baseUrl}/remote_entry.js`;
+                        remotes[product.name] = `${product.name}@${product.baseUrl}/remote_entry.js`;
+                    } else {
+                        console.log(`Product ${product.name} not found`);
+                    }
+                }
             } else {
-                console.log(`Product ${product.name} not found`);
+                // For production, hardcode the URLs of product containers to be based on the web app URL
+                for (const product of products) {
+                    remotes[product.name] = `${product.name}@[window.basename]/static/products/${product.name}/remote_entry.js`;
+                }
+            }
+        } else {
+            console.warn('Skipping initialization of products');
+        }
 
-                // Add false aliases to prevent Webpack from trying to resolve the missing modules
-                aliases[product.name] = false;
-                aliases[`${product.name}/manifest`] = false;
+        const aliases = {};
+
+        for (const product of products) {
+            if (remotes[product.name]) {
+                continue;
             }
+
+            // Add false aliases to prevent Webpack from trying to resolve the missing modules
+            aliases[product.name] = false;
+            aliases[`${product.name}/manifest`] = false;
```

Comment:
neat , i didnt knew it would ignore if we passed false to alias

Reply:
Yeah, that makes it so that Webpack doesn't try to bundle that file at build time. I don't know what would happen if we tried to import it at runtime, but that's not an issue since we never call that because `REMOTE_CONTAINERS` is sent to empty below this